app/assets/javascripts/app/supported-states.js

```diff
@@ -0,0 +1,39 @@
+function validateStateSelection() {
```

Comment:
This is difficult to review without the corresponding changes to add the data attributes. What do you think about including these bits with the rails code that does the work to add those?

Reply:
Let me take a look. I tried once and it spiraled out pretty bad, but I'll give it another go.